app/views/rides/_form.html.haml

```diff
@@ -56,12 +64,19 @@
 
   .col-sm-8.field
     = f.label :pickup_at, "#{h t(:what_time_pickup)} <sup>*</sup>".html_safe
-    = datetime_select("ride", "pickup_at", selected: Date.new(2018,11,06), start_year: 2018, end_year: 2018, ampm: true)
+    - if @ride.persisted?
+      = datetime_select("ride", "pickup_at", selected: @ride.pickup_in_time_zone, ampm: true)
+    - else
+      = datetime_select("ride", "pickup_at", selected: Date.new(2018,11,06), start_year: 2018, end_year: 2018, start_hour: 6, ampm: true)
```

Comment:
You can collapse the `if` if you initialize the pickup date to `Date.new(2018,11,06)` in the controller.